TVRemoteControl/Networking/Protocols/RequestDispatcher.swift

```diff
@@ -0,0 +1,21 @@
+import Foundation
+
+protocol RequestDispatching {
+    typealias RequestDispatcherCompletion = (Data?, Error?, HTTPURLResponse?) -> Void
```

Comment:
From my perspective, the completion with this approach (with optional data/error) is not as useful as the completion with `Swift.Result`.

```swift
switch result {
case .success(let data):
 // do something
case .failure(let error):
  // handle error  
} 
```


Could you replace it?

Reply:
it is actually just a completion of `URLSession.dataTask`. You can check how it is used in `HTTPRequestDispatcher`.
`HTTPRequestDispatcher` is used inside `URLSessionEngine` and `URLSessionEngine`'s completion handler is Swift `Result`. This way  `RequestDispatching` only cares about how to "dispatch" request and doesn't really care about response handling, handling of result and build final completion are responsibilities of `URLSessionEngine` aka `APIRequesting`.